tests/test_bvoc.F90

```diff
@@ -0,0 +1,166 @@
+
+   !call Emis_Allocate(GridState, EmisState, RC) !Not sure why cannot call it even if I made it public and used EmisStateMod in this module
+   if (EmisState%nCats > 0) then
+      do c = 1, EmisState%nCats
+         do s = 1, EmisState%Cats(c)%nSpecies
+            ALLOCATE(EmisState%Cats(c)%Species(s)%Flux(GridState%number_of_levels), STAT=RC)
+            if (RC /= CC_SUCCESS) then
+               ErrMsg = 'Error allocating "EmisState%Cats%Species%Flux"!'
+               call cc_emit_error(ErrMsg, RC, ThisLoc)
+               stop 1  !!Note here is not 'return'
+            endif
+         end do
+      end do
+   end if
+
+   !----------------------------
+   ! Test 2
+   !----------------------------
+
+   ! Meteorological State (get from 20190620 18:00:00 HEMCO log out)
+   MetState%TSTEP= 3600
+   MetState%LAI= 3.9094312191009521
+   allocate(MetState%PFT_16(16))
+   MetState%PFT_16= (/0.00, 0.11120668053627014, 0.00, 0.00, 0.00, 0.00, 0.00, 0.35108909010887146, &
+      0.00, 0.00, 0.00, 0.00, 0.00,0.18369837105274200,6.9862455129623413E-002,0.26875913143157959/)
+   MetState%PMISOLAI= 3.90559316  ! needs to multiply PFTSUM
+   MetState%Q_DIR_2= 368.02691650390625
+   MetState%Q_DIFF_2= 55.688854217529297
+   MetState%PARDR_LASTXDAYS= 69.8014755
+   MetState%PARDF_LASTXDAYS= 37.4157791
+   MetState%TS= 300.42892456054688
+   MetState%T_LASTXDAYS= 294.497833
+   MetState%T_LAST24H= 294.919861
+   MetState%GWETROOT= 0.76981580257415771
+   MetState%SUNCOS= 0.96700188446067026
+   MetState%LAT=  38.00
+   MetState%DOY= 171
+   MetState%LocalHour= 12.00
+   MetState%D_BTW_M=  1.00
+   !These are read from file. Keep them here for now
+   !MetState%AEF_ISOP= 1.8055753025901623E-009
+   !MetState%AEF_MBOX= 4.9856540616165277E-013
+   !MetState%AEF_BPIN= 1.3127712426530056E-011
+   !MetState%AEF_CARE= 3.6563258621377127E-012
+   !MetState%AEF_LIMO= 6.3985420662872528E-012
+   !MetState%AEF_OCIM= 3.0705341449874024E-011
+   !MetState%AEF_SABI= 1.0054971341792413E-011
+
+   ! Allocate DiagState
+   call cc_allocate_diagstate(Config, DiagState, ChemState, RC)
+   if (rc /= CC_SUCCESS) then
+      errMsg = 'Error in cc_allocate_diagstate'
+      stop 1
+   endif
+
+   title = "BVOC Test 2 | Test each species"
+   Config%bvoc_activate = .TRUE.
+
+   !call cc_bvoc_init(Config, ChemState, EmisState, BvocState, RC)
+   call cc_bvoc_init(Config, EmisState, BvocState, RC)
+   if (rc /= CC_SUCCESS) then
+      errMsg = 'Error in cc_bvoc_init'
+      call cc_emit_error(errMsg, rc, thisLoc)
+      stop 1
+   end if
+
+   !call cc_bvoc_run(MetState, EmisState, DiagState, BvocState, ChemState, RC )
+   call cc_bvoc_run(MetState, EmisState,  DiagState, BvocState, RC )
+   if (rc /= CC_SUCCESS) then
+      errMsg = 'Error in cc_bvoc_run'
+      call cc_emit_error(errMsg, rc, thisLoc)
+      stop 1
+   end if
+
+   call print_info(Config, BvocState, MetState, DiagState, title)
+   call assert(BvocState%TotalEmission > 0.0_fp, "Test BVOC species")
```

Comment:
we should also add a check to make sure it is a reasonable value.  ie ```fortran 
assert(BvocState%ISOP > 50) 
assert(BvocState%ISOP < 100) 
```

note the limits here are arbitrary

Reply:
@lwcugb